process/block/metablock.go

```diff
@@ -1294,3 +1297,28 @@ func (mp *metaProcessor) IsInterfaceNil() bool {
 	}
 	return false
 }
+
+func (mp *metaProcessor) getShardHeaderWithNonce(shardId uint32, nonce uint64) (*block.Header, []byte, error) {
```

Comment:
this code is similar to getMetaHeaderWithNonce

it looks like we can move this function to baseProcess and maybe give as parameter the pool while returning a HeaderHandler.
I think we can then do the type assertion after call and have a single function to handle both.

Reply:
I found the real problem, and I think I will remove this methods anyway